spec/requirements.md

```diff
@@ -43,3 +44,15 @@ Ideally the entire API could be free of side effects and odd conventions, like m
 
 The system should support exporting generics and classes. If classes are objects, they can be treated like any other object when exporting and importing symbols. If generics are objects, then it should be simple to export all methods on a generic.  It is not clear whether selective method export is important. One use case would be to hide a method with an internal class in its signature to avoid unnecessary documentation. Perhaps `R CMD check` could ignore methods for unexported classes. There should be no need for explicit method registration.  
 
+## Reflection and dynamism
+
+Given a class and a generic, you should be able to find the appropriate method without calling it. This is important for building tooling around the system.
+
+On the flip side, you should also be able to register a method dynamically at run-time. This is important for:
```

Comment:
Add "lazily" and mention classes in addition to methods, since the same problem occurs when extending a class in another package.

Reply:
Done — clarified the other bullets a little more too.